File: src/main/java/com/examples/testing/withkotlinfauxproperty/Book.java
package com.examples.testing.withkotlinfauxproperty;

import edu.illinois.cs.cs125.questioner.lib.Correct;

/*
 * Question to test detection of Kotlin faux properties in Jenisol 2023.7.0.
 */
@Correct(name = "Kotlin Faux Property Testing", version = "2023.7.0", author = "challen@illinois.edu")
public class Book {
  private String name;

  public void setName(String setName) {
    name = setName;
  }

  public String getName() {
    return name;
  }
}

File: src/main/java/com/examples/testing/withkotlinfauxproperty/correct/kotlin/Book.kt
package com.examples.testing.withkotlinfauxproperty.correct.kotlin

import edu.illinois.cs.cs125.questioner.lib.AlsoCorrect

/*
 * Question to test detection of Kotlin faux properties in Jenisol 2023.7.0.
 */

@AlsoCorrect
class Book {
  var name: String? = null
}